cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java

```diff
@@ -176,17 +178,47 @@ private void ensureJDBCDriverIsAvailable() throws Exception {
       LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
                 dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, driverClass.getName(),
                 JDBCDriverShim.class.getName());
-      driverShim = new JDBCDriverShim(driverClass.newInstance());
+      if (driverShim == null) {
+        driverShim = new JDBCDriverShim(driverClass.newInstance());
```

Comment:
`driverClass` is null if this method is called from `tableNameExists()` (which in turn call from `configurePipeline()`, which doesn't set any value to the `drvierClass` field).

I prefer not to have `driverClass` as a field, but rather just a parameter passed to this method.